src/main/java/org/openstreetmap/atlas/geography/atlas/items/TurnRestriction.java

```diff
@@ -239,6 +240,21 @@ private TurnRestriction(final Relation relation)
             // Try to re-build the route, based on the "from", "via" and "to" members of the
             // relation
 
+            // According to OSM Wiki a restriction relation member can not have more than 1 via node
+            // https://wiki.openstreetmap.org/wiki/Relation:restriction#Members
+            // If the relation has more than 1 via node then discard the restriction as it is
+            // incorrectly modeled.
+            // To bring back the turn restriction OSM data needs to be modeled correctly
+            final long viaNodeCount = relation.members().stream().filter(
+                    member -> "via".equals(member.getRole()) && member.getEntity() instanceof Node)
+                    .count();
+            if (viaNodeCount > 1)
+            {
+                throw new CoreException(
+                        "Restriction relation should not have more than 1 via node. But, {} has {} via nodes",
+                        relation.getOsmIdentifier(), viaNodeCount);
```

Comment:
Instead of iterating through the relation members twice, maybe we can move this check to after the viaMembers have been populated and iterate through viaMembers instead?

Reply:
Plus one on this and it would simplify the condition to check only for Nodes, since at this point we're only looking at the via roles!

Reply:
Makes sense 👍